.github/workflows/_build.yml

```diff
@@ -175,10 +174,76 @@ jobs:
       - name: Run Tests
         run: ./scripts/ci ${{ matrix.schemes }} test
 
+  Screenshots:
+    name: Screenshots tests
+    runs-on: macos-latest
+    needs: [Setup]
+    permissions:
+      pull-requests: write
+      contents: write
+    timeout-minutes: 20
+    continue-on-error: true
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Cache Bundler install Gems
+        uses: actions/cache@v4
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ env.ImageVersion }}-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-gems-${{ env.ImageVersion }}
+
+      - name: Cache Pods
+        uses: actions/cache@v4
+        with:
+          path: Example/Pods
+          key: ${{ runner.os }}-pods-${{ hashFiles('**/Podfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-pods-
+
+      - name: Bundle Install
+        run: bundle install --jobs 4 --retry 3
+
+      - name: Pod Install
+        run: bundle exec pod install
+        working-directory: Example
+
+      - name: Select Xcode 14
+        run: sudo xcode-select --switch /Applications/Xcode_14.2.app
+      
+      - name: Screenshot Tests
+        id: screenshotTests
+        run: |
+          ./scripts/record_screenshot_tests.sh
+
+          changedFiles=`git status --porcelain` && echo "CHANGED_FILES=${changedFiles//$'\n'/'%0A'}" >> $GITHUB_OUTPUT
+        continue-on-error: true
+
+      - name: Check snapshot changes
+        id: checkSnapshotChanges
+        if: ${{ github.event_name != 'pull_request' }}
+        run: ./scripts/lint
+
+      - name: Commit changes
```

Comment:
another thing to consider as an option which is something i wanna consider on android as well: you could have two modes: by default you just use the snapshot tests as tests (ie just verify nothing has changed) and share the test output in a nice format - most snapshot testing libraries have that by default. then you can add a trigger for actually re-recording the snapshots if intentional. that way you get fewer unintentional changes - especially with flaky changes.
again depends on your tooling, duration etc - on android we switched to the current approach as the tests were very slow, so waiting for the initial failure to then trigger the actual change would take forever, but it also results in that confusing behaviour where because the screenshots changed your CI status is pending for all steps. Now that snapshot tests are a lot faster & stable the standard approach of treating CI as a test by default might be better again